### PR TITLE
Fixed infinite loop in cluster validation.

### PR DESCRIPTION
**Pull request by jszczepkowski**
Fixed infinite loop in cluster validation when getting nodes failed.

### CODE REVIEW

File: cluster/validate-cluster.sh [hunk 1]
```diff
@@ -91,7 +93,12 @@ while true; do
   # Suppress errors from kubectl output because during cluster bootstrapping
   # for clusters where the master node is registered, the apiserver will become
   # available and then get restarted as the kubelet configures the docker bridge.
-  node=$(kubectl_retry get nodes) || continue
+  node=$(kubectl_retry get nodes) || if [[ "${attempt}" -gt "${MAX_ATTEMPTS}" ]]; then
```
Comment (mwielgus) [inline cluster/validate-cluster.sh]: Can you make this simpler - for example don't mix assignment and if with exit and continue. Its hard to tell what this thing is exactly doing without good bash knowledge.

Comment (jszczepkowski) [inline cluster/validate-cluster.sh]: I did some refactoring, but I'm not sure if it is much better.

**Comment by mwielgus** [comment]
LGTM

**Comment by saad-ali** [comment]
Cherry-picked to 1.5 in 39137

**Pull request merged**